src/equations/compressible_navier_stokes_1d.jl

```diff
@@ -0,0 +1,466 @@
+    rho, rho_v1, _ = u
+
+    v1 = rho_v1 / rho
+    T = temperature(u, equations)
+
+    return SVector(gradient_entropy_vars[1],
+                   T * (gradient_entropy_vars[2] + v1 * gradient_entropy_vars[3]), # grad(u) = T*(grad(w_2)+v1*grad(w_3))
+                   T * T * gradient_entropy_vars[3])
+end
+
+# This routine is required because `prim2cons` is called in `initial_condition`, which
+# is called with `equations::CompressibleEulerEquations1D`. This means it is inconsistent
+# with `cons2prim(..., ::CompressibleNavierStokesDiffusion1D)` as defined above.
+# TODO: parabolic. Is there a way to clean this up?
+@inline function prim2cons(u, equations::CompressibleNavierStokesDiffusion1D)
+    prim2cons(u, equations.equations_hyperbolic)
+end
+
+@inline function temperature(u, equations::CompressibleNavierStokesDiffusion1D)
+    rho, rho_v1, rho_e = u
+
+    p = (equations.gamma - 1) * (rho_e - 0.5 * rho_v1^2 / rho)
+    T = p / rho
+    return T
+end
+
+# TODO: can we generalize this to MHD?
+"""
+    struct BoundaryConditionNavierStokesWall
+
+Creates a wall-type boundary conditions for the compressible Navier-Stokes equations.
+The fields `boundary_condition_velocity` and `boundary_condition_heat_flux` are intended
+to be boundary condition types such as the `NoSlip` velocity boundary condition and the
+`Adiabatic` or `Isothermal` heat boundary condition.
+
+!!! warning "Experimental feature"
+    This is an experimental feature and may change in future releases.
+"""
+struct BoundaryConditionNavierStokesWall{V, H}
+    boundary_condition_velocity::V
+    boundary_condition_heat_flux::H
+end
+
+"""
+    struct NoSlip
+
+Use to create a no-slip boundary condition with `BoundaryConditionNavierStokesWall`. The field `boundary_value_function`
+should be a function with signature `boundary_value_function(x, t, equations)`
+and should return a `SVector{NDIMS}` whose entries are the velocity vector at a
+point `x` and time `t`.
+"""
+struct NoSlip{F}
+    boundary_value_function::F # value of the velocity vector on the boundary
+end
+
+"""
+    struct Isothermal
+
+Used to create a no-slip boundary condition with [`BoundaryConditionNavierStokesWall`](@ref).
+The field `boundary_value_function` should be a function with signature
+`boundary_value_function(x, t, equations)` and return a scalar value for the
+temperature at point `x` and time `t`.
+"""
+struct Isothermal{F}
+    boundary_value_function::F # value of the temperature on the boundary
+end
+
+"""
+    struct Adiabatic
+
+Used to create a no-slip boundary condition with [`BoundaryConditionNavierStokesWall`](@ref).
+The field `boundary_value_normal_flux_function` should be a function with signature
+`boundary_value_normal_flux_function(x, t, equations)` and return a scalar value for the
+normal heat flux at point `x` and time `t`.
+"""
+struct Adiabatic{F}
+    boundary_value_normal_flux_function::F # scaled heat flux 1/T * kappa * dT/dn
+end
```

Comment:
Aren't these structs already defined in the file for the 2D system? Since the structs are generic, could we please move them to another file - maybe `src/equations/compressible_navier_stokes.jl`?

Reply:
Yes and no - I moved them from 2D to 1D for now. Adding a seperate file `src/equations/compressible_navier_stokes.jl` is a more expressive way, I think